pkg/profilecreator/profilecreator.go

```diff
@@ -281,18 +282,12 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 		return "", "", fmt.Errorf("Error determining if Hyperthreading is enabled or not: %v", err)
 	}
 
-	//TODO: Make the allocation logic below more readable by using separate helper functions, one per allocation strategy
-	// (splitReservedCPUsAcrossNUMA=false/true -> two strategies) each one with its clear and nice allocation loop
-	for numaID, node := range topologyInfo.Nodes {
-		if splitReservedCPUsAcrossNUMA {
-			if remainder != 0 {
-				max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder)
-				remainder--
-			} else {
-				max = max + reservedPerNuma
-			}
+	for numaID, node := range topologyInfoNodes {
```

Comment:
if we add the `totalCPUSetFromTopology` helper I suggest below, we can reason about the expanded cpu list here and probably use a clearer logic. Let's explore.

Reply:
see below. Scratch this.

Reply:
Ack